### PR TITLE
Use correct IP for rate limiting

### PR DESCRIPTION
**Pull request by zephraph**
Last minute fix to make sure we're getting the ip address from the right place for rate limiting

### CODE REVIEW

File: src/lib/middleware/rateLimiting.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export const rateLimiterMiddlewareFactory = redisClient => {
       return
     }
     rateLimiter
-      .consume(req.connection.remoteAddress)
+      .consume(req.headers["x-forwarded-for"] || req.connection.remoteAddress)
```
Comment (sweir27) [inline src/lib/middleware/rateLimiting.ts]: Do we need the `|| req.connection.remoteAddress` if we're always serving from behind a proxy?

Comment (zephraph) [reply src/lib/middleware/rateLimiting.ts]: Probably not. But if it's not there for whatever reason, we'd have a fall back. It shouldn't ever get triggered though, so I _assume_ it's safe, right?

Comment (l2succes) [reply src/lib/middleware/rateLimiting.ts]: `X-Forwarded-For` can return multiple IP addresses in comma delimited format, we might need to handle that gracefully as well

Comment (zephraph) [reply src/lib/middleware/rateLimiting.ts]: Oh? Do I need to split it on commas?

Comment (l2succes) [reply src/lib/middleware/rateLimiting.ts]: Maybe we should use https://www.npmjs.com/package/request-ip. it handles all common edge cases (based on different) proxies. From docs:

> The user ip is determined by the following order:
> 
> 1. X-Client-IP
> 2. X-Forwarded-For (Header may return multiple IP addresses in the format: "client IP, proxy 1 IP, proxy 2 IP", so we take the the first one.)
> 3. CF-Connecting-IP (Cloudflare)
> 4. Fastly-Client-Ip (Fastly CDN and Firebase hosting header when forwared to a cloud function)
> 5. True-Client-Ip (Akamai and Cloudflare)
> 6. X-Real-IP (Nginx proxy/FastCGI)
> 7. X-Cluster-Client-IP (Rackspace LB, Riverbed Stingray)
> 8. X-Forwarded, Forwarded-For and Forwarded (Variations of #2)
> 9. req.connection.remoteAddress
> 10. req.socket.remoteAddress
> 11. req.connection.socket.remoteAddress
> 12. req.info.remoteAddress

Comment (sweir27) [reply src/lib/middleware/rateLimiting.ts]: Looking to see  if the lib we're using has support for this...

Comment (zephraph) [reply src/lib/middleware/rateLimiting.ts]: kk

Comment (l2succes) [reply src/lib/middleware/rateLimiting.ts]: But yeah, if we stick to just using `X-Forwarded-for`

```javascript
app.get(/, function (req, res){
   var forwardedIpsStr = req.header('x-forwarded-for');
   var IP = '';

   if (forwardedIpsStr) {
      IP = forwardedIps = forwardedIpsStr.split(',')[0];  
   }
});
```

Comment (izakp) [reply src/lib/middleware/rateLimiting.ts]: Express' `req.ip` should do this already

**Pull request merged**